labeling/service.go

```diff
@@ -129,33 +147,181 @@ func (s *Server) wantAnyRecords(ctx context.Context, ra *events.RepoAppend) bool
+	// images
+	if blob.MimeType == "image/png" || blob.MimeType == "image/jpeg" {
+		// only an image API is configured
+		if s.muNSFWImgLabeler != nil || s.hiveAILabeler != nil {
+			return true
+		}
+	}
+	return false
+}
+
 func (s *Server) labelRecord(ctx context.Context, did, nsid, uri, cid string, rec cbg.CBORMarshaler) ([]string, error) {
 	log.Infof("labeling record: %v", uri)
 	var labelVals []string
+	var blobs []lexutil.Blob
 	switch nsid {
 	case "app.bsky.feed.post":
 		post, suc := rec.(*appbsky.FeedPost)
 		if !suc {
 			return []string{}, fmt.Errorf("record failed to deserialize from CBOR: %s", rec)
 		}
+		/* XXX(bnewbold): debugging broken post record
+		postJson, _ := json.Marshal(post)
+		log.Infof("labeling post: %v", string(postJson))
+		buf := new(bytes.Buffer)
+		post.MarshalCBOR(buf)
+		log.Infof("post CBOR: %v", buf.Bytes())
+		*/
+
 		// run through all the keyword labelers on posts, saving any resulting labels
-		for _, labeler := range s.kwl {
+		for _, labeler := range s.kwLabelers {
 			for _, val := range labeler.LabelPost(*post) {
 				labelVals = append(labelVals, val)
 			}
 		}
+
+		if s.sqrlLabeler != nil {
+			sqrlVals, err := s.sqrlLabeler.LabelPost(ctx, *post)
+			if err != nil {
+				return []string{}, fmt.Errorf("failed to label post with SQRL: %v", err)
+			}
+			labelVals = append(labelVals, sqrlVals...)
+		}
+
+		// record any image blobs for processing
+		if post.Embed != nil && post.Embed.EmbedImages != nil {
+			for _, eii := range post.Embed.EmbedImages.Images {
+				blobs = append(blobs, *eii.Image)
+			}
+		}
 	case "app.bsky.actor.profile":
 		profile, suc := rec.(*appbsky.ActorProfile)
 		if !suc {
 			return []string{}, fmt.Errorf("record failed to deserialize from CBOR: %s", rec)
 		}
+		// XXX(bnewbold): debugging broken profile record
+		profileJson, _ := json.Marshal(profile)
+		log.Infof("labeling profile: %v", string(profileJson))
+		buf := new(bytes.Buffer)
+		profile.MarshalCBOR(buf)
+		log.Infof("profile CBOR: %v", buf.Bytes())
+
 		// run through all the keyword labelers on posts, saving any resulting labels
-		for _, labeler := range s.kwl {
-			for _, val := range labeler.LabelActorProfile(*profile) {
+		for _, labeler := range s.kwLabelers {
+			for _, val := range labeler.LabelProfile(*profile) {
 				labelVals = append(labelVals, val)
 			}
 		}
+
+		if s.sqrlLabeler != nil {
+			sqrlVals, err := s.sqrlLabeler.LabelProfile(ctx, *profile)
+			if err != nil {
+				return []string{}, fmt.Errorf("failed to label profile with SQRL: %v", err)
+			}
+			labelVals = append(labelVals, sqrlVals...)
+		}
+
+		// record avatar and/or banner blobs for processing
+		if profile.Avatar != nil {
+			blobs = append(blobs, *profile.Avatar)
+		}
+		if profile.Banner != nil {
+			blobs = append(blobs, *profile.Banner)
+		}
+	}
+
+	log.Infof("will process %d blobs", len(blobs))
+	for _, blob := range blobs {
+		if blob.Cid == "" {
+			return []string{}, fmt.Errorf("received stub blob (CID undefined)")
```

Comment:
any particular reason not to just return nil? In an error case you arent going to use the value

Reply:
updated this and similar cases